cypress-tests/cypress/e2e/adyen-card-flow-e2e-test.cy.ts

```diff
@@ -0,0 +1,72 @@
+import * as testIds from "../../../src/Utilities/TestUtils.bs";
+import {
+  getClientURL,
+  createPaymentBody,
+  changeObjectKeyValue,
+} from "../support/utils";
+
+describe("Adyen Card payment flow test", () => {
+  const publishableKey = Cypress.env("HYPERSWITCH_PUBLISHABLE_KEY");
+  const secretKey = Cypress.env("HYPERSWITCH_SECRET_KEY");
+  let getIframeBody: () => Cypress.Chainable<JQuery<HTMLBodyElement>>;
+  const iframeSelector =
+    "#orca-payment-element-iframeRef-orca-elements-payment-element-payment-element";
+  const adyenIframeSelector = ".adyen-checkout__iframe--threeDSIframe";
+
+  changeObjectKeyValue(
+    createPaymentBody,
+    "profile_id",
+    Cypress.env("PROFILE_ID")
+  );
+  changeObjectKeyValue(createPaymentBody, "authentication_type", "three_ds");
+
+  beforeEach(() => {
+    getIframeBody = () => cy.iframe(iframeSelector);
+    cy.createPaymentIntent(secretKey, createPaymentBody).then(() => {
+      cy.getGlobalState("clientSecret").then((clientSecret) => {
+        cy.visit(getClientURL(clientSecret, publishableKey));
+      });
+    });
+  });
+
+  it("title rendered correctly", () => {
+    cy.contains("Hyperswitch Unified Checkout").should("be.visible");
+  });
+
+  it("orca-payment-element iframe loaded", () => {
+    cy.frameLoaded(iframeSelector);
+  });
+
+  it("submit payment form and make successful payment", () => {
+    cy.clickElementInIframe(testIds.addNewCardIcon);
+
+    cy.fixture("adyen3DSCardDetails").then((cardDetails) => {
+      cy.enterValueInIframe(testIds.cardNoInputTestId, cardDetails.cardNumber);
+      cy.enterValueInIframe(testIds.expiryInputTestId, cardDetails.expiryDate);
+      cy.enterValueInIframe(testIds.cardCVVInputTestId, cardDetails.cvv);
+    });
+
+    cy.intercept("**/payments/redirect/**").as("hyperswitchRedriect");
+    cy.intercept("**/checkoutshopper/threeDS2.shtml*").as("adyenCheckout");
```

Comment:
```suggestion
```

Reply:
intercepting the redirection and wait is a good thing as we do not know when the page will load